### PR TITLE
LPD-22768 refactored timeline modal and dropdown options

### PR DESCRIPTION
**Pull request by ctang3**
https://liferay.atlassian.net/browse/LPD-22768

### CODE REVIEW

File: modules/apps/change-tracking/change-tracking-web/src/main/java/com/liferay/change/tracking/web/internal/display/context/ViewTimelineHistoryDisplayContext.java [hunk 1]
```diff
@@ -31,14 +33,29 @@ public ViewTimelineHistoryDisplayContext(
...
 		_themeDisplay = (ThemeDisplay)renderRequest.getAttribute(
 			WebKeys.THEME_DISPLAY);
 	}
 
 	public String getAPIURL() {
-		return "/o/change-tracking-rest/v1.0/ct-entries/history";
+		long classNameId = ParamUtil.getLong(_renderRequest, "classNameId");
+		long classPK = ParamUtil.getLong(_renderRequest, "classPK");
+
+		StringBundler queryParamsSB = new StringBundler(4);
```
Comment (david-truong) [inline modules/apps/change-tracking/change-tracking-web/src/main/java/com/liferay/change/tracking/web/internal/display/context/ViewTimelineHistoryDisplayContext.java:47]: bchan will probably prefer you just name queryParamsSB to sb

**Comment by ctang3** [comment]
ci:test:sf

**Comment by ctang3** [comment]
ci:test:publications

**Comment by ctang3** [comment]
ci:test:relevant

**Comment by david-truong** [comment]
@ctang3 can you take a look at the failures

**Comment by ctang3** [comment]
ci:test:publications

**Comment by david-truong** [comment]
ci:test:sf

**Comment by ctang3** [comment]
ci:test:sf

**Comment by ctang3** [comment]
Pushed a fix for the missing "Edit in Publication" itemAction in the modal. The code that changed is:

- Addition of a "checkout" permissionKey in [CTEntryResourceImpl](https://github.com/liferay-site-management/liferay-portal/blob/daa5742af03f7dd4f3fd4dee67b1ef980a1a2dd3/modules/apps/change-tracking/change-tracking-rest-impl/src/main/java/com/liferay/change/tracking/rest/internal/resource/v1_0/CTEntryResourceImpl.java#L213-L230). The other FDS examples for checking out a CTCollection were called in a FDS that returns CTCollections so when I applied it to this code, it didn't work because this history endpoint returns CTEntries. So it checks against CTEntryResourceImpl instead of CTCollectionResourceImpl.
- Changed the [href for the checkout FDSActionDropdownItem in ViewTimelineHistoryDisplayContext ](https://github.com/liferay-site-management/liferay-portal/blob/daa5742af03f7dd4f3fd4dee67b1ef980a1a2dd3/modules/apps/change-tracking/change-tracking-web/src/main/java/com/liferay/change/tracking/web/internal/display/context/ViewTimelineHistoryDisplayContext.java#L65-L83)to call javascript `window.open(...)` with the target set to `_top` so that the parent page refreshes and the user is placed in the selected Publication. Without this change, only the iframe refreshes - the parent window doesn't refresh so the user remains in the initial Publication unless they manually refresh or navigate elsewhere.

**Comment by ctang3** [comment]
ci:test:publications

**Comment by ctang3** [comment]
ci:test:stop

**Comment by ctang3** [comment]
ci:stop

**Comment by ctang3** [comment]
ci:test:publications

**Comment by david-truong** [comment]
ci:test:sf

**Comment by david-truong** [comment]
ci:test:publications

**Comment by ctang3** [comment]
Taking a look

**Comment by ctang3** [comment]
ci:test:sf

**Comment by ctang3** [comment]
ci:test:publications

**Comment by ctang3** [comment]
ci:test:publications

**Comment by ctang3** [comment]
ci:test:publications

**Comment by david-truong** [comment]
ci:test:relevant

**Comment by david-truong** [comment]
ci:test:sf

**Comment by david-truong** [comment]
ci:report:27522519

**Comment by david-truong** [comment]
ci:forward

**Comment by david-truong** [comment]
ci:forward

**Comment by david-truong** [comment]
ci:forward:force

**Pull request closed without merging**